lib/helpers/getMessage.js

```diff
@@ -13,18 +12,21 @@ var isHeader = /^HEADER/g;
  * }
  *
  * @param {object} message an ImapMessage from the node-imap library
+ * @param {string} encoding if set then the bodies will be decoded and returned as buffers
  * @returns {Promise} a promise resolving to `message` with schema as described above
  */
-module.exports = function getMessage(message) {
+module.exports = function getMessage(message, encoding) {
     return new Promise(function (resolve) {
         var attributes;
         var messageParts = [];
+        var isHeader = /^HEADER/g;
 
         function messageOnBody(stream, info) {
             var body = '';
 
             function streamOnData(chunk) {
-                body += chunk.toString('utf8');
+	        // Default is just use utf8
+                body += encoding ? chunk.toString() : chunk.toString('utf8');
```

Comment:
`chunk.toString() & chunk.toString('utf8')` [do the same thing](https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end)